esmerald/encoders.py

```diff
@@ -1,102 +1,70 @@
 from __future__ import annotations
 
-from typing import Any, TypeVar, get_args
+from typing import Any, get_args
 
-import msgspec
-from lilya._internal._encoders import json_encoder as json_encoder  # noqa
-from lilya._utils import is_class_and_subclass
+from lilya._internal._encoders import ModelDumpEncoder
 from lilya.encoders import (
-    ENCODER_TYPES as LILYA_ENCODER_TYPES,  # noqa
-    Encoder as LilyaEncoder,  # noqa
-    register_encoder as register_encoder,  # noqa
+    ENCODER_TYPES as ENCODER_TYPES_CTX,  # noqa
+    Encoder,
+    EncoderProtocol,  # noqa
+    MoldingProtocol,  # noqa
+    json_encode,
+    register_encoder,  # noqa
 )
-from msgspec import Struct
-from pydantic import BaseModel
-from pydantic_core import PydanticSerializationError
 
 from esmerald.exceptions import ImproperlyConfigured
 from esmerald.utils.helpers import is_union
 
-T = TypeVar("T")
+ENCODER_TYPES = ENCODER_TYPES_CTX.get()
 
-ENCODER_TYPES = LILYA_ENCODER_TYPES.get()
 
-
-class Encoder(LilyaEncoder[T]):
-    def is_type(self, value: Any) -> bool:
-        """
-        Function that checks if the function is
-        an instance of a given type
-        """
-        raise NotImplementedError("All Esmerald encoders must implement is_type() method.")
-
-    def serialize(self, obj: Any) -> Any:
-        """
-        Function that transforms a data structure into a serializable
-        object.
-        """
-        raise NotImplementedError("All Esmerald encoders must implement serialize() method.")
-
-    def encode(self, annotation: Any, value: Any) -> Any:
-        """
-        Function that transforms the kwargs into a structure
-        """
-        raise NotImplementedError("All Esmerald encoders must implement encode() method.")
-
-
-class MsgSpecEncoder(Encoder):
-    def is_type(self, value: Any) -> bool:
-        return isinstance(value, Struct) or is_class_and_subclass(value, Struct)
-
-    def serialize(self, obj: Any) -> Any:
-        """
-        When a `msgspec.Struct` is serialised,
-        it will call this function.
-        """
-        return msgspec.json.decode(msgspec.json.encode(obj))
-
-    def encode(self, annotation: Any, value: Any) -> Any:
-        return msgspec.json.decode(msgspec.json.encode(value), type=annotation)
+def register_esmerald_encoder(
+    encoder: EncoderProtocol | MoldingProtocol | type[EncoderProtocol] | type[MoldingProtocol],
+) -> None:
+    """
+    Registers an esmerald encoder into available Lilya encoders
+    """
+    try:
+        register_encoder(encoder)
+    except RuntimeError:
+        raise ImproperlyConfigured(f"{type(encoder)} must be a subclass of Encoder") from None
 
 
-class PydanticEncoder(Encoder):
-    def is_type(self, value: Any) -> bool:
-        return isinstance(value, BaseModel) or is_class_and_subclass(value, BaseModel)
+PydanticEncoder = ModelDumpEncoder
+try:
```

Comment:
NO need to use the try here. Msgspec is a hard dependency for now. Rollback this